apps/avatax/src/pages/api/webhooks/order-confirmed.ts

```diff
@@ -87,14 +91,41 @@ export default wrapWithLoggerContext(
           return webhookResponse.success();
         }
 
-        // TODO: Map errors
         if (taxProviderResult.isErr()) {
-          logger.error("Error confirming order", { error: taxProviderResult.error });
-
-          return webhookResponse.error(taxProviderResult.error);
+          const error = taxProviderResult.error;
+
+          logger.debug("Error confirming order", { error });
+
+          switch (true) {
+            case error instanceof ActiveConnectionServiceErrors.WrongChannelError: {
+              /**
+               * Subscription can listen on every channel or no channels.
+               * However, app works only for some of them (which are configured to be used with taxes routing).
+               * If this happens, webhook will be received, but this is no-op.
+               */
+              return res.status(202).send("Channel not configured with the app.");
+            }
+
+            case error instanceof ActiveConnectionServiceErrors.MissingMetadataError:
+            case error instanceof ActiveConnectionServiceErrors.ProviderNotAssignedToChannelError:
+            case error instanceof ActiveConnectionServiceErrors.BrokenConfigurationError: {
+              return res.status(400).send("App is not configured properly.");
+            }
+            case error instanceof ActiveConnectionServiceErrors.MissingChannelSlugError: {
+              return res
+                .status(500)
+                .send("Webhook didn't contain channel slug. This should not happen.");
+            }
+            default: {
+              Sentry.captureException(taxProviderResult.error);
+              logger.fatal("Unhandled error", { error });
+
+              return res.status(500).send("Unhandled error");
+            }
```

Comment:
Have you tried something like this?
```
    if (error instanceof BaseError) {
      const errorName = error.name as keyof ActiveConnectionServiceErrors;

      switch (errorName) {
        case "MissingChannelSlugError":
          return false;
        case "MissingMetadataError": {
          return true;
        }
        default:
          const x: never = errorName;

          return x;
      }
    }
```

Reply:
Yea good idea, modern-errors indeed keep internal name (but its still leaky, because `keyof ActiveConnectionServiceErrors` is not tightly bound with error name)

Reply:
I'd keep it like it is for, now extracting strings safe-manner will add even more boilerplate. I will proceed with more refactors in the future